src/cpp/core/CMakeLists.txt

```diff
@@ -282,3 +281,35 @@ target_link_libraries(
    ${Boost_LIBRARIES}
    ${CORE_SYSTEM_LIBRARIES}
 )
+
+# define executable (for running unit tests)
+if (RSTUDIO_UNIT_TESTS_ENABLED)
+
+  file(GLOB_RECURSE CORE_TEST_FILES "*Tests.cpp")
+
+  add_executable(rstudio-core-bin
+     TestMain.cpp
+     ${CORE_TEST_FILES}
+     ${CORE_HEADER_FILES}
+   )
+
+   set_target_properties(rstudio-core-bin
+      PROPERTIES
+         OUTPUT_NAME rstudio-core
+   )
+
+   target_link_libraries(rstudio-core-bin
+      rstudio-core
+      rstudio-core-synctex
+      rstudio-core-hunspell
+      ${Boost_LIBRARIES}
+      ${CORE_SYSTEM_LIBRARIES}
+   )
+   
+   # run tests after build
```

Comment:
I don't think we should run the tests after every build as this will quickly limit us to tests that don't run for more than (in total) ~ 3 seconds. We can use the --test argument along with a script like we do for other binaries (e.g. ./rserver-dev or ./rstudio-dev).